crates/primitives/src/request.rs

```diff
@@ -23,3 +25,25 @@ impl IntoIterator for Requests {
         self.0.into_iter()
     }
 }
+
+impl Encodable for Requests {
+    fn encode(&self, out: &mut dyn bytes::BufMut) {
+        self.0
+            .iter()
+            .map(Encodable7685::encoded_7685)
+            .map(Bytes::from)
+            .collect::<Vec<_>>()
+            .encode(out)
```

Comment:
I don't think we need the alloc here and can just mimic the vec encode impl

Reply:
need an alloc regardless, made it more cursed in [7500152](https://github.com/paradigmxyz/reth/pull/8271/commits/7500152ce2a6448191da74f3b53c028cb4d6d237), feel free to give it a shot